sdk/tables/data-tables/MigrationGuide.md

```diff
@@ -0,0 +1,387 @@
+
+To improve the development experience across Azure services, a set of uniform [design guidelines](https://azure.github.io/azure-sdk/general_introduction.html) was created for all languages to drive a consistent experience with established API patterns for all services. A set of [TypeScript & JavaScript Guidelines](https://azure.github.io/azure-sdk/typescript_introduction.html) was also introduced to ensure that TypeScript clients have a natural and idiomatic feel with respect to the TypeScript and JavaScript ecosystems. The new `@azure/data-tables` follows these guidelines.
+
+### Cross Service SDK improvements
+
+The modern `@azure/data-tables` client library also provides the ability to share in some of the cross-service improvements made to the Azure development experience, such as
+
+- Using the new `@azure/identity` library to share a single authentication approach between clients
+- A unified logging and diagnostics pipeline offering a common view of the activities across each of the client libraries
+- Use of promises rather than callbacks for a simplified programming experience
+- Use of async iterators in paging APIs
+
+## Important changes
+
+### Package name
+
+The modern client library is named `@azure/data-tables` and was released beginning with version 12. The legacy client library is named `azure-storage` with version of 2.x.x or below.
+
+### Constructing the clients
+
+Previously in `azure-storage`, you would use `createTableService` which can be used to get an instance of the `TableService` in order to perform service level operations
+
+```javascript
+const azure = require('azure-storage');
+var tableService = azure.createTableService("<connection-string>");
+```
+
+Now, in `@azure/data-tables`, we need a TableServiceClient for service level operations.
+```javascript
+const { TableServiceClient } = require('@azure/data-tables');
+const tableService = TableServiceClient.fromConnectionString("<connection-string>")
+```
+
+### Creating a Table
+Previously in `azure-storage`, we'd use a `TableService` instance to create a table. The `createTable` method would take a callback to execute once the table has been created. This forces sequential operations to be inside the callback, potentially creating a callback chain
+
+```javascript 
+const azure = require('azure-storage');
+var tableService = azure.createTableService("<connection-string>");
+
+const tableName = "<table-name>"
+tableService.createTable(tableName);
+
+```
+
+With `@azure/data-tables` we have access to all table level operations directly from the `TableServiceClient`. Because the table service client is not affinitized to any one table, it is ideal for scenarios where you need to create, delete, or list more than one Table.
```

Comment:
```suggestion
With `@azure/data-tables` we have access to all table level operations directly from the `TableServiceClient`. Because the table service client is not affinitized to any one table, it is ideal for scenarios where you need to create, delete, or list more than one table.
```